src/sif/autocost.cc

```diff
@@ -80,6 +80,8 @@ constexpr ranged_default_t<float> kUseDistanceRange{0, kDefaultUseDistance, 1.0f
 constexpr ranged_default_t<float> kAutoHeightRange{0, kDefaultAutoHeight, 10.0f};
 constexpr ranged_default_t<float> kAutoWidthRange{0, kDefaultAutoWidth, 10.0f};
 constexpr ranged_default_t<uint32_t> kProbabilityRange{0, kDefaultRestrictionProbability, 100};
+constexpr ranged_default_t<uint32_t> kFixedSpeedRange{0, 0, baldr::kMaxSpeedKph};                                                    
```

Comment:
overall I think you should move the whole logic into the parent class in `sif/dynamiccost`, similar to how it's done for `top_speed`. that'll remove a lot of redundant code. e.g. move this line here to `dynamiccost.cc:108`. I'll give some tips where to move the other parts as well.

Reply:
think you already did that after reviewing everything. so you mostly need to remove some stuff